iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -362,16 +362,30 @@ public StorageFormatDescriptor getStorageFormatDescriptor(org.apache.hadoop.hive
     return null;
   }
 
-  public boolean supportsAppendData(org.apache.hadoop.hive.metastore.api.Table table) throws SemanticException {
+  public boolean supportsAppendData(org.apache.hadoop.hive.metastore.api.Table table, boolean withPartClause)
+      throws SemanticException {
     Table icebergTbl = IcebergTableUtil.getTable(conf, table);
-    return icebergTbl.spec().isUnpartitioned();
```

Comment:
could we return right away if the table is Unpartitioned to skip the manifests listing

Reply:
yep, done